src/Config/ConfigReader.php

```
@@ -152,6 +157,45 @@ private function parseYamlFile(string $file): array
+
+        $reflection = new ReflectionObject($b);
+        $parameters = $reflection->getConstructor()?->getParameters() ?? [];
+        $properties = [];
+
+        foreach ($parameters as $parameter) {
+            $property = $reflection->getProperty($parameter->getName());
+            $aValue = $property->getValue($a);
+            $bValue = $property->getValue($b);
+
+            /* @phpstan-ignore notEqual.notAllowed (Loose comparison is intended as we may compare objects) */
+            if ($parameter->getDefaultValue() != $bValue) {
```

PHPStan (2.2.6)

[error] No error with identifier notEqual.notAllowed is reported on line 184.


(ignore.unmatchedIdentifier)